tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -233,11 +219,38 @@ internal void WriteClosedSpans()
 
             if (spansToWrite.Count > 0)
             {
+                _ = GetSamplingPriority(triggerSamplingDecision: true);
+
                 RunSpanSampler(spansToWrite);
                 Tracer.Write(spansToWrite);
             }
         }
 
+        /// <summary>
+        /// Gets the trace's sampling priority, optionally triggering a sampling decision.
+        /// </summary>
+        public int? GetSamplingPriority(bool triggerSamplingDecision = false)
```

Comment:
I still think two different methods would be better, something like this 😛 

```csharp
        public int? TryGetSamplingPriority()
        public int GetOrMakeSamplingPriority()
```

Reply:
> In particular, if the trace context doesn't contain any spans, we can't make a sampling decision yet because at the very minimum, we need a trace id

To be honest, I didn't realise that you could _have_ a `TraceContext` without a `Span`. Like... does that make sense even? 🤔 

The only cases I can see is when you're manually creating a _parent_ span context, but then that feels different as well, because I don't know that you _should_ be able to call `GetOrMakeSamplingPriority` on that sort of context 🤔

Reply:
> I didn't realise that you could have a `TraceContext` without a `Span`

I think it can happen in those special cases where we create spans by calling `Tracer.StartSpan(..., addToTraceContext = false)`. This code path creates the `TraceContext` if needed, but doesn't call `TraceContext.AddSpan(Span)`. So `span.SpanContext.TraceContext` is available, but that `TraceContext` could be empty (no spans) until we add this "orphan" span to it or create new spans. It's... weird, and I would be happy to somehow get rid of that edge case.

Reply:
>  creating a parent span context, but then that feels different as well

Yes, that's a "stand-alone" span context not connected to local trace (`SpanContext.TraceContext` will be `null`). Those can carry their own sampling priority (`SpanContext.SamplingPriority`, not `TraceContext.SamplingPriority`), but that should only be used for propagation purposes.

`<cough>`_span context refactor_`</cough>`

Reply:
In that case, feel free to ignore my suggestions, seems like they should just be part of a larger refactor later.

> It's... weird, and I would be happy to somehow get rid of that edge case.

Good lord, yes. 😅 It does feel like this needs an overhaul...

> <cough>span context refactor</cough>

_pleeeeease_